### PR TITLE
(DOCSP-26906): Add note about unsupported Sync progress notifications to SDKs

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-26906

### Staged Changes

- [Java SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26906/sdk/java/examples/sync-changes-between-devices/#check-upload---download-progress-for-a-sync-session)
- [.NET SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26906/sdk/dotnet/sync/sync-progress/#monitor-sync-progress)
- [Node.js SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26906/sdk/node/examples/sync-changes-between-devices/#check-upload---download-progress-for-a-sync-session)
- [React Native SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26906/sdk/react-native/sync-data/check-upload-download-progress/)
- [Swift SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26906/sdk/swift/sync/sync-progress/)

I've omitted this note from Flutter while waiting to hear back whether it's supported there, and from Kotlin & C++ because neither has this API implemented.

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/includes/flex-sync-unsupported-progress-notifications.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.. note::
+
+   Flexible Sync progress notifications are not yet fully supported. Downloads
+   only report notifications after changes are integrated. Partition-Based Sync
+   provides ongoing notifications as changes progress.
```
Comment (mongodben) [inline source/includes/flex-sync-unsupported-progress-notifications.rst:5]: would be good to mention that uploads work as expected for flex sync here

**Review by mongodben — COMMENTED** [review]
one small suggestion abt explicitly mentioning uploads.

also could you add content to flutter page per nikola's comment? 

i think it'd probably just be a sentence change. but if larger amnt of work and you don't cover in this PR, we can create separate ticket for it.

**Comment by dacharyc** [comment]
> also could you add content to flutter page per nikola's comment?
> 
> i think it'd probably just be a sentence change. but if larger amnt of work and you don't cover in this PR, we can create separate ticket for it.

Sure, could do, @mongodben, but I didn't add anything to Flutter here because you already sort of have something that - while not explicitly detailing his note - sort of makes this one a moot point anyway? The bit about the 16MB increments at the bottom of the "Monitor Sync Progress" section: https://www.mongodb.com/docs/realm/sdk/flutter/sync/manage-sync-session/#monitor-sync-progress

I think the other SDKs are expecting something like continuous updates, so not seeing them has been problematic to some devs, but if the expectation here is already for updates to be sporadic, I'm not sure we need to get into more details. Do you still think we should add something?

File: source/includes/flex-sync-unsupported-progress-notifications.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.. note::
+
+   Flexible Sync progress notifications are not yet fully supported. Downloads
+   only report notifications after changes are integrated. Partition-Based Sync
+   provides ongoing notifications as changes progress. Upload progress should
+   report regularly for both Sync Modes.
```
Comment (mongodben) [inline source/includes/flex-sync-unsupported-progress-notifications.rst:6]:
```suggestion
   Flexible Sync progress notifications are not yet fully supported.
   When using Flexible Sync, downloads only report notifications after changes are integrated.
   Partition-Based Sync provides ongoing notifications as changes progress downloading. 
   Upload progress reports regularly for both Sync Modes.
```

added some clarification which i think is important, even if copy a little more verbose. what you think?

Comment (dacharyc) [reply source/includes/flex-sync-unsupported-progress-notifications.rst:6]: Clarity seems important in this case. Taking it!

File: source/includes/flex-sync-unsupported-progress-notifications.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.. note::
+
+   Flexible Sync progress notifications are not yet fully supported.
+   When using Flexible Sync, downloads only report notifications after changes are integrated.
+   Partition-Based Sync provides ongoing notifications as changes progress downloading. 
+   Upload progress reports regularly for both Sync Modes.
```
Comment (mongodben) [inline source/includes/flex-sync-unsupported-progress-notifications.rst:6]:
```suggestion
   Uploads report ongoing progress notifications for both Sync Modes.
```

rephrased to not use word 'regularly', which i think you meant as 'in regular increments' but could be misconstrued to mean 'normally'.

**Comment by mongodben** [comment]
> Sure, could do, @mongodben, but I didn't add anything to Flutter here because you already sort of have something that - while not explicitly detailing his note - sort of makes this one a moot point anyway? The bit about the 16MB increments at the bottom of the "Monitor Sync Progress" section: https://www.mongodb.com/docs/realm/sdk/flutter/sync/manage-sync-session/#monitor-sync-progress

i think it'd be beneficial to be more explicit about it. could be another bullet point in the section with the 16MB one, and clarify the 16MB bullet point. 

those bullets could be refactored to smthn like:

```rst
- When monitoring downloads, the number of transferred and transferable bytes are only estimates.
  The downloaded Device Sync changesets are compressed with
  `gzip <https://www.gnu.org/software/gzip/>`_ before transmitting, so
  the actual size of transmitted bytes will be smaller than the reported number
  of both transferable and transferred bytes.
- Realm optimizes download speeds by combining multiple changesets
  into a single download message, up to 16 MB. Since the progress callback is
  only invoked once before and after a download message is processed, this
  means that you'll likely see ``transferredBytes`` change in increments of roughly
  16 MB when monitoring download progress rather than continuously
  as the message is being downloaded.
-  Uploads report ongoing progress notifications. The changes are not zipped.
```

**Comment by mongodben** [comment]
per [eng feedback on slack](https://mongodb.slack.com/archives/CL9JSLTGE/p1671028778964609), let's not touch flutter stuff here. 

tyler said he made a ticket for this, and it'll require a non-trivial refactor of that section.

don't wanna block this PR which is relevant for the other SDKs.

**Review by mongodben — APPROVED** [review]
lgtm!

**Pull request merged**